src/usersyncroniser.ts

```diff
@@ -140,9 +140,31 @@ export class UserSyncroniser {
         }
     }
 
-    public async EnsureJoin(member: GuildMember, roomId: string) {
+    public async JoinRoom(member: GuildMember, roomId: string) {
         const state = await this.GetUserStateForGuildMember(member, "");
-        log.info(`Ensuring ${state.id} is joined to ${roomId}`);
+        log.info(`Joining ${state.id} in ${roomId}`);
+        try {
+            await this.ApplyStateToRoom(state, roomId, member.guild.id);
+        } catch (e) {
+            if (e.errcode !== "M_FORBIDDEN") {
+                log.error(`Failed to join ${state.id} to ${roomId}`, e);
+                throw e;
+            } else {
+                log.info(`User not in room ${roomId}, inviting`);
+                try {
+                    await this.bridge.getIntent().invite(roomId, state.mxUserId);
+                    await this.ApplyStateToRoom(state, roomId, member.guild.id);
+                } catch (e) {
+                    log.error(`Failed to join ${state.id} to ${roomId}`, e);
+                    throw e;
```

Comment:
Does this not get caught in the surrounding catch one level below, and therefore JoinRoom will not throw (which ideally it should)

Reply:
we need it to throw so that the schedule joiner in matrixRoomHandler can do its job

Reply:
Sure but if the user fails to join, then this function won't throw and that's bad. You have a catch inside a catch.

Reply:
It should still throw, as it is in the catch block, not the try block.

```
try {
   throw "blubb"; // falls back to catch
} catch (e) {
    throw e; // throws an outermost error
}
```

and (labeling try-1 and try-2 incorrectly for easier meaning stuffs)

```
try-1 {
    throw "blubb"; // throws error, landing in catch-1
} catch-1 (e) {
    try-2 {
        throw "blubb"; // throws an error, landing in catch-2
    } catch-2 (e) {
        throw e; // this is outside of any try.....so throw it all the way
    }
}
```

Reply:
Yes sorry I'm having a senior moment. That makes sense :)